app/ducks/walletActions.js

```diff
@@ -368,12 +368,31 @@ async function parseInputsOutputs(net, tx) {
       covValue += output.value;
     }
 
+    if (covenant.action == 'FINALIZE') {
+
+      // isSender if there exists an input with value 0 and own address
+      const isSender = tx.inputs.findIndex(x => x.value === 0 && x.path !== null) !== -1;
```

Comment:
Why `value === 0` here? TRANSFER and FINALIZE outputs may have a value (the second highest bid, the amount of the "burned" coins by auction winner). So if I am the original name owner, the finalize will have one input from my wallet address (the TRANSFER) which will have a `value >= 0` depending on the auction. I think you may have to look for a TRANSFER covenant in the input and check if its a wallet address (`path !== null`)

Reply:
Oh I see, hadn't thought of that. The inputs in the tx object only has address and path, and no covenants so did this to identify the input with the name.
To test this, I transferred a name that was finalized with payment so it's like `TRANSFER` -> `FINALIZE(pay)` and now `TRANSFER` -> `FINALIZE(pay)` again.

**2nd transfer** (finalized with 30 HNS)
Sender side:
```
inputs: Array(2)
0: {value: 0, address: "rs1q084zn6eawcvrhkt8v2agrywwwfc5u58flx9q3j", path: {…}}
1: {value: 0, address: "rs1qhhu0qhfwx8kl9qvprqn274w6v58msp490dy8ug", path: null}

outputs: Array(3)
0: {value: 0, address: "rs1q5uy4kg8ts6z5qzlmaazqywr4hrdv3tkz725v86", covenant: {…}, path: null}
1: {value: 19981740, address: "rs1qemjhwa032430m6e4wss0ycvrehytfk0j4jqkl6", covenant: {…}, path: null}
2: {value: 30000000, address: "rs1q6yrp0tvksy8espkjwcfwxc845sk0wdtyuh6ceu", covenant: {…}, path: {…}}
```

Receiver side:
```
inputs: Array(2)
0: {value: 0, address: "rs1q084zn6eawcvrhkt8v2agrywwwfc5u58flx9q3j", path: null}
1: {value: 49988200, address: "rs1qhhu0qhfwx8kl9qvprqn274w6v58msp490dy8ug", path: {…}}

outputs: Array(3)
0: {value: 0, address: "rs1q5uy4kg8ts6z5qzlmaazqywr4hrdv3tkz725v86", covenant: {…}, path: {…}}
1: {value: 19981740, address: "rs1qemjhwa032430m6e4wss0ycvrehytfk0j4jqkl6", covenant: {…}, path: {…}}
2: {value: 30000000, address: "rs1q6yrp0tvksy8espkjwcfwxc845sk0wdtyuh6ceu", covenant: {…}, path: null}
```

The name input value is always 0, and the values are calculated and shown as +/- 30 HNS properly.
I guess this works because there's no value for `TRANSFER`s, and `FINALIZE` can only happen after a `TRANSFER`.

Reply:
Are you testing the swaps with a name you "won" on regtest with only one bid? Because in that scenario the names value would be 0.

In [this atomic swap tx for the name `pinkieswear`](https://hnsnetwork.com/txs/32227eb2dab16dbb3debc873ff40cfeba299eb2cf1554ffd098fa968bc88c6cd) you can see the FINALIZE value is 4 HNS.

Input 0 to that transaction spends [this TRANSFER output](https://hnsnetwork.com/txs/294ba9d77ee9c09d5f385e33772370608a0884d46f143d85017ce62bb2678ae4) which also has a value of 4 HNS.

Reply:
Okay tried it with multiple bids and yeah the input has value (but only when path is also available, 0 otherwise).
So will have to find out how to get covenant info for inputs also to identify the one with names.

Reply:
As per chat on telegram, updated and doesn't depend on name value anymore.